apollo-federation/src/query_plan/query_planning_traversal.rs

```diff
@@ -84,8 +87,9 @@ pub(crate) struct QueryPlanningTraversal<'a, 'b> {
     /// True if query planner `@defer` support is enabled and the operation contains some `@defer`
     /// application.
     has_defers: bool,
-    /// The initial fetch ID generation (used when handling `@defer`).
-    starting_id_generation: u64,
+    /// A handle to the sole generator of fetch IDs. While planning an operation, only one of
+    /// generator can be used.
+    id_generator: Arc<FetchIdGenerator>,
```

Comment:
This was causing a defer test to fail because some fetch nodes has overlapping IDs. Now, there is a single "global" ID generator used for the entire planning process. While looking at some of the test, I noticed a side effect of this. Sometime, an id will be generated but never needed/used. This can cause holes in the IDs used in the first QP.

I point this out because the only invariant around fetch IDs mentioned is that they must be unique. However, I'm unsure if the router expects fetch IDs to not have holes. If this is the case, I can rework this solution.